@mfasDa the errors are genuine:
```
FAILED: Detectors/EMCAL/workflow/CMakeFiles/O2lib-EMCALWorkflow.dir/src/OfflineCalibSpec.cxx.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DBOOST_ALL_NO_LIB -DBOOST_CONTAINER_DYN_LINK -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_REGEX_DYN_LINK -DBOOST_SERIALIZATION_DYN_LINK -DDPL_ENABLE_BACKTRACE -DENABLE_UPGRADES -DFAIRMQ_HAS_STD_FILESYSTEM=1 -DFMT_LOCALE -DFMT_SHARED -DGENERATORS_WITH_HEPMC3 -DGENERATORS_WITH_PYTHIA6 -DGENERATORS_WITH_PYTHIA8 -DO2lib_EMCALWorkflow_EXPORTS -DZDC_FASTSIM_ONNX -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Detectors/EMCAL/workflow/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/BUILD/772a98f94e4b7cf4806cdc9a89030d539c2c47bb/O2/Detectors/EMCAL/workflow -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/BUILD/772a98f94e4b7cf4806cdc9a89030d539c2c47bb/O2/Detectors/EMCAL/workflow/src -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Framework/Core/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Common/Utils/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/DataFormats/common/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Common/Constants/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/GPU/Common -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Framework/Logger/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Common/MathUtils/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/DataFormats/Headers/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/DataFormats/MemoryResources/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/CCDB/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Framework/Foundation/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Framework/Foundation/3rdparty/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Utilities/PCG/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/DataFormats/Detectors/CTP/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/DataFormats/simulation/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/DataFormats/Detectors/Common/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/GPU/Utils -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Utilities/rANS/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Detectors/Base/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/DataFormats/Reconstruction/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Common/Field/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/DataFormats/Parameters/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Common/Types/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Common/SimConfig/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/DataFormats/Detectors/EMCAL/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Detectors/EMCAL/simulation/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Detectors/EMCAL/base/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Detectors/Raw/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Detectors/EMCAL/reconstruction/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Steer/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Generators/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Framework/Utils/include -I/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Algorithm/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/Common-O2/v1.6.0-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/boost/v1.75.0-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/Configuration/v2.6.3-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/libInfoLogger/v2.5.1-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/Monitoring/v3.12.0-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/curl/7.70.0-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/FairMQ/v1.4.52-local1/include/fairmq -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/FairMQ/v1.4.52-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/FairLogger/v1.11.1-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/fmt/8.1.1-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/ROOT/v6-26-04-patches-alice1-local4/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/FairRoot/v18.4.8-local4/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/VMC/v1-1-p1-local4/include/vmc -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/arrow/v9.0.0-alice1-local2/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/Vc/1.4.1-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/ms_gsl/4.0.0-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/libjalienO2/0.1.3-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/RapidJSON/v1.1.0-alice2-local3/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/libuv/v1.40.0-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/TBB/v2021.5.0-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/pythia/v8304-local1/include -isystem /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/osx_arm64/HepMC3/3.2.5-local4/include -fPIC -O2 -std=c++17 -Werror -Wno-error=deprecated-declarations -O2 -g -Wno-unknown-warning-option -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk -mmacosx-version-min=10.15 -fPIC -std=gnu++17 -MD -MT Detectors/EMCAL/workflow/CMakeFiles/O2lib-EMCALWorkflow.dir/src/OfflineCalibSpec.cxx.o -MF Detectors/EMCAL/workflow/CMakeFiles/O2lib-EMCALWorkflow.dir/src/OfflineCalibSpec.cxx.o.d -o Detectors/EMCAL/workflow/CMakeFiles/O2lib-EMCALWorkflow.dir/src/OfflineCalibSpec.cxx.o -c /System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Detectors/EMCAL/workflow/src/OfflineCalibSpec.cxx
/System/Volumes/Data/build/nomad/alloc/2a93ae05-6ab2-ab1b-aeeb-44de2ea51f23/ci/local/o2/sw/SOURCES/O2/9650/0/Detectors/EMCAL/workflow/src/OfflineCalibSpec.cxx:62:15: error: no member named 'getCellEnergy' in 'o2::emcal::Cell'; did you mean 'getEnergy'?
        if (c.getCellEnergy() > 0.5) {
```

--- a/Detectors/EMCAL/workflow/CMakeLists.txt
+++ b/Detectors/EMCAL/workflow/CMakeLists.txt
@@ -20,6 +20,7 @@ o2_add_library(EMCALWorkflow
                        src/AnalysisClusterSpec.cxx
                        src/RawToCellConverterSpec.cxx
                        src/EntropyEncoderSpec.cxx
+                       src/OfflineCalibSpec.cxx
                        src/EntropyDecoderSpec.cxx
                        src/StandaloneAODProducerSpec.cxx
                PUBLIC_LINK_LIBRARIES O2::Framework O2::DataFormatsCTP O2::DataFormatsEMCAL O2::EMCALSimulation O2::Steer
@@ -52,6 +53,13 @@ o2_add_executable(cell-writer-workflow
                   COMPONENT_NAME emcal
                   SOURCES src/cell-writer-workflow.cxx
                   PUBLIC_LINK_LIBRARIES O2::EMCALWorkflow)
+
+o2_add_executable(emc-offline-calib-workflow
+                  COMPONENT_NAME emcal
+                  SOURCES src/emc-offline-calib-workflow.cxx
+                  PUBLIC_LINK_LIBRARIES O2::Framework
+                                        O2::EMCALWorkflow
+                                        O2::EMCALCalibration)
 
 o2_add_executable(cell-reader-workflow
                   COMPONENT_NAME emcal

--- a/Detectors/EMCAL/workflow/include/EMCALWorkflow/OfflineCalibSpec.h
+++ b/Detectors/EMCAL/workflow/include/EMCALWorkflow/OfflineCalibSpec.h
@@ -1,0 +1,73 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include <string>
+#include "Framework/DataProcessorSpec.h"
+#include "Framework/Task.h"
+
+#include "TFile.h"
+#include "TH1.h"
+#include "TH2.h"
+#include "THnSparse.h"
+
+namespace o2
+{
+
+namespace emcal
+{
+
+/// \class OfflineCalibSpec
+/// \brief Task for producing offline calibration objects
+/// \ingroup EMCALworkflow
+/// \author Hannah Bossi <hannah.bossi@cern.ch>, Yale University
+/// \since August 16th, 2022
+///
+/// This task fills offline calibration objects for the EMCAL.
+class OfflineCalibSpec : public framework::Task
+{
+ public:
+  /// \brief Constructor
+  /// \param makeCellIDTimeEnergy If true the THnSparseF of cell ID, time, and energy is made
+  OfflineCalibSpec(bool makeCellIDTimeEnergy) : mMakeCellIDTimeEnergy(makeCellIDTimeEnergy){};
+
+  /// \brief Destructor
+  ~OfflineCalibSpec() override = default;
+
+  /// \brief Initializing the offline calib task
+  /// \param ctx Init context
+  void init(framework::InitContext& ctx) final;
+
+  /// \brief Fill histograms needed for the offline calibration
+  /// \param ctx Processing context
+  ///
+  void run(framework::ProcessingContext& ctx) final;
+
+  /// \brief Write histograms to an output root file
+  /// \param ec end of stream context
+  ///
+  void endOfStream(o2::framework::EndOfStreamContext& ec) final;
+
+ private:
+  std::unique_ptr<TH2> mCellAmplitude;         ///< Cell energy vs. cell ID
+  std::unique_ptr<TH2> mCellTime;              ///< Cell time vs. cell ID
+  std::unique_ptr<TH1> mNevents;               ///< Number of events
+  std::unique_ptr<THnSparseF> mCellTimeEnergy; ///< ID, time, energy
+  bool mMakeCellIDTimeEnergy = true;           ///< Switch whether or not to make a THnSparseF of cell ID, time, and energy
+};
+
+/// \brief Creating offline calib spec
+/// \ingroup EMCALworkflow
+///
+o2::framework::DataProcessorSpec getEmcalOfflineCalibSpec(bool makeCellIDTimeEnergy);
+
+} // namespace emcal
+
+} // namespace o2

--- a/Detectors/EMCAL/workflow/src/OfflineCalibSpec.cxx
+++ b/Detectors/EMCAL/workflow/src/OfflineCalibSpec.cxx
@@ -1,0 +1,95 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include <iostream>
+#include <vector>
+#include <type_traits>
+#include <gsl/span>
+
+#include "FairLogger.h"
+
+#include "Framework/ControlService.h"
+#include "Framework/DataRefUtils.h"
+#include "DataFormatsEMCAL/Cell.h"
+#include "DataFormatsEMCAL/TriggerRecord.h"
+#include "EMCALWorkflow/OfflineCalibSpec.h"
+
+using namespace o2::emcal;
+
+void OfflineCalibSpec::init(o2::framework::InitContext& ctx)
+{
+  // energy vs. cell ID
+  mCellAmplitude = std::unique_ptr<TH2>(new TH2F("mCellAmplitude", "Cell amplitude", 800, 0., 40., 17664, -0.5, 17663.5));
+  // time vs. cell ID
+  mCellTime = std::unique_ptr<TH2>(new TH2F("mCellTime", "Cell time", 800, -200, 600, 17664, -0.5, 17663.5));
+  // number of events
+  mNevents = std::unique_ptr<TH1>(new TH1F("mNevents", "Number of events", 1, 0.5, 1.5));
+  if (mMakeCellIDTimeEnergy) {
+    // cell time, cell energy, cell ID
+    std::array<int, 3> arrNBins = {17664, 100, 100}; // NCells, time, energy
+    std::array<double, 3> arrMin = {-0.5, -50, 0};
+    std::array<double, 3> arrMax = {17663.5, 50., 50};
+    mCellTimeEnergy = std::unique_ptr<THnSparseF>(new THnSparseF("CellIDvsTimevsEnergy", "CellIDvsTimevsEnergy", arrNBins.size(), arrNBins.data(), arrMin.data(), arrMax.data()));
+  }
+}
+
+void OfflineCalibSpec::run(framework::ProcessingContext& pc)
+{
+  auto cells = pc.inputs().get<gsl::span<o2::emcal::Cell>>("cells");
+  auto triggerrecords = pc.inputs().get<gsl::span<o2::emcal::TriggerRecord>>("triggerrecord");
+  LOG(debug) << "[EMCALOfflineCalib - run] received " << cells.size() << " cells from " << triggerrecords.size() << " triggers ...";
+  if (triggerrecords.size()) {
+    for (const auto& trg : triggerrecords) {
+      if (!trg.getNumberOfObjects()) {
+        LOG(debug) << "[EMCALOfflineCalib - run] Trigger does not contain cells, skipping ...";
+        continue;
+      }
+      LOG(debug) << "[EMCALOfflineCalib - run] Trigger has " << trg.getNumberOfObjects() << " cells  ..." << std::endl;
+      gsl::span<const o2::emcal::Cell> objectsTrigger(cells.data() + trg.getFirstEntry(), trg.getNumberOfObjects());
+      for (const auto& c : objectsTrigger) {
+        LOG(debug) << "[EMCALOfflineSpec - run] Channel: " << c.getTower();
+        LOG(debug) << "[EMCALOfflineSpec - run] Energy: " << c.getEnergy();
+        LOG(debug) << "[EMCALOfflineSpec - run] Time: " << c.getTimeStamp();
+        mCellAmplitude->Fill(c.getEnergy(), c.getTower());
+        if (c.getEnergy() > 0.5) {
+          mCellTime->Fill(c.getTimeStamp(), c.getTower());
+          if (mMakeCellIDTimeEnergy) {
+            mCellTimeEnergy->Fill(c.getTower(), c.getTimeStamp(), c.getEnergy());
+          }
+        }
+      }
+      mNevents->Fill(1);
+    }
+  }
+}
+
+void OfflineCalibSpec::endOfStream(o2::framework::EndOfStreamContext& ec)
+{
+  // write histograms to root file here
+  std::unique_ptr<TFile> outputFile(new TFile("emcal-offline-calib.root", "RECREATE"));
+  outputFile->cd();
+  mCellAmplitude->Write();
+  mCellTime->Write();
+  mNevents->Write();
+  if (mMakeCellIDTimeEnergy) {
+    mCellTimeEnergy->Write();
+  }
+  outputFile->Close();
+}
+
+o2::framework::DataProcessorSpec o2::emcal::getEmcalOfflineCalibSpec(bool makeCellIDTimeEnergy)
+{
+  return o2::framework::DataProcessorSpec{"EMCALOfflineCalib",
+                                          {{"cells", o2::header::gDataOriginEMC, "CELLS", 0, o2::framework::Lifetime::Timeframe},
+                                           {"triggerrecord", o2::header::gDataOriginEMC, "CELLSTRGR", 0, o2::framework::Lifetime::Timeframe}},
+                                          {},
+                                          o2::framework::adaptFromTask<o2::emcal::OfflineCalibSpec>(makeCellIDTimeEnergy)};
+}

--- a/Detectors/EMCAL/workflow/src/emc-offline-calib-workflow.cxx
+++ b/Detectors/EMCAL/workflow/src/emc-offline-calib-workflow.cxx
@@ -1,0 +1,41 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "EMCALWorkflow/OfflineCalibSpec.h"
+#include "CommonUtils/ConfigurableParam.h"
+#include "Framework/ConfigParamSpec.h"
+
+using namespace o2::framework;
+
+// ------------------------------------------------------------------
+
+// we need to add workflow options before including Framework/runDataProcessing
+void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
+{
+  // option allowing to set parameters
+  std::vector<ConfigParamSpec> options{{"makeCellIDTimeEnergy", VariantType::Bool, true, {"list whether or not to make the cell ID, time, energy THnSparse"}},
+                                       {"configKeyValues", VariantType::String, "", {"Semicolon separated key=value strings"}}};
+  workflowOptions.insert(workflowOptions.end(), options.begin(), options.end());
+}
+
+// ------------------------------------------------------------------
+
+#include "Framework/runDataProcessing.h"
+
+WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
+{
+  WorkflowSpec wf;
+  // Update the (declared) parameters if changed from the command line
+  bool makeCellIDTimeEnergy = cfgc.options().get<bool>("makeCellIDTimeEnergy");
+  o2::conf::ConfigurableParam::updateFromString(cfgc.options().get<std::string>("configKeyValues"));
+  wf.emplace_back(o2::emcal::getEmcalOfflineCalibSpec(makeCellIDTimeEnergy));
+  return wf;
+}

